### PR TITLE
[Bug][docs] Fix wrong links in README.md

### LINKED ISSUE #10393: [Bug][docs] Some links in README.md is wrong
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

```markdown
## Compile and install

See [Compilation](https://doris.apache.org/docs/install/source-install/compilation-with-ldb-toolchain.html)

## Getting start

See [Basic Usage](https://doris.apache.org/docs/data-table/basic-usage.html)
```

The above links in `README.md` is invalid.

### What You Expected?

Fix the wrong links.

### How to Reproduce?

Just click these links in homepage.

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by adonis0147**
# Proposed changes

Issue Number: close #10393 

## Problem Summary:

Fix wrong links in README.md.

## Checklist(Required)

1. Does it affect the original behavior: No
2. Has unit tests been added: No Need
3. Has document been added or modified: No Need
4. Does it need to update dependencies: No
5. Are there any changes that cannot be rolled back: No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,11 +55,11 @@ Doris mainly integrates the technology of [Google Mesa](https://research.google/
 
 ## Compile and install
 
-See [Compilation](https://doris.apache.org/install/source-install/compilation-with-ldb-toolchain.html)
+See [Compilation](https://doris.apache.org/docs/install/source-install/compilation-with-ldb-toolchain.html)
```
[the remark this answers is not in the document]
Comment (Hisoka-X) [reply README.md:58]: Maybe use compilation.html as link better, because this is compile's first sub page

[the remark this answers is not in the document]
Comment (adonis0147) [reply README.md:58]: Hi @Hisoka-X , I think it is better to recommend novices compile the codebase by [ldb toolchain](https://github.com/amosbird/ldb_toolchain_gen) for the following reasons:
1. It is difficult for users without root privilege to deploy `docker` on server.
2. It is easy to install [ldb toolchain](https://github.com/amosbird/ldb_toolchain_gen) and it is validated even on Centos 6.
3. [Dockerfile](https://github.com/apache/doris/blob/master/docker/Dockerfile) provided by Doris also uses [ldb toolchain](https://github.com/apache/doris/blob/9036f93df4a54e218c0ccb3ff2496bce39f5099d/docker/Dockerfile#L45) to build the codebase.

[the remark this answers is not in the document]
Comment (Hisoka-X) [reply README.md:58]: I got it. Thanks for reply.

**Review by yangzhg — DISMISSED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]
LGTM

**Pull request merged**